controllers/cloud.redhat.com/clowdapp_controller.go

```diff
@@ -427,21 +428,25 @@ func contains(list []string, s string) bool {
 	return false
 }
 
-func (r *ClowdAppReconciler) runProviders(provider *providers.Provider, a *crd.ClowdApp) error {
+func (r *ClowdAppReconciler) runProviders(log logr.Logger, provider *providers.Provider, a *crd.ClowdApp) error {
```

Comment:
It seems like you already get this from the struct on line 77. What's the advantage of passing this in vs using `r.Log....`?

Reply:
log context